utils/src/main/java/software/amazon/awssdk/utils/AttributeMap.java

```diff
@@ -327,17 +328,17 @@ private void internalPut(Key<?> key, Value<?> value) {
          */
         private Value<?> internalComputeIfAbsent(Key<?> key, Supplier<Value<?>> value) {
             checkCopyOnUpdate();
-            return attributes.compute(key, (k, v) -> {
-                if (v == null || resolveValue(v) == null) {
-                    Value<?> newValue = value.get();
-                    Validate.notNull(newValue, "Supplied value must not be null.");
-                    if (v != null) {
-                        dependencyGraph.valueUpdated(v, newValue);
-                    }
-                    return newValue;
+            Value<?> currentValue = attributes.get(key);
```

Comment:
Looks good. Is there any way in which we tested this ? If a unit test is difficult, A one-time runnable test that validates these changes would be helpful here. Lmk if its non-trivial to test / validate

Reply:
This new behavior is exercised in https://github.com/aws/aws-sdk-java-v2/pull/5562. Let me see if I can create a standalone test for it, for the purposes of this PR.